dataset_conversion/convert_bids_to_nnUnetv2.py

```diff
@@ -50,11 +52,71 @@ def get_parser():
+        counter (int): Number of subject already treated.
+        list_images (list): List with all the images path in the new dataset.
+        list_labels (list): List with all the labels path in the new dataset.
+        is_ses (bool): Is the dataset session structured.
+        session (str): Session name or None if dataset without session .
+
+    Returns:
+        list_images (list): List with all the images path in the new dataset with one new append.
+        list_labels (list): List with all the labels path in the new dataset with one new append.
+
+    """
+    if is_ses:
+        # TODO verify that using _ between subject and session is problematic or not
+        subject_image_file = os.path.join(root, subject, session, 'anat',
+                                          f"{subject}_{session}_{contrast}.nii.gz")
+        subject_label_file = os.path.join(root, 'derivatives', 'labels', subject, session, 'anat',
+                                          f"{subject}_{session}_{contrast}_{label_suffix}.nii.gz")
+        # TODO use regex
+        sub_name = str(Path(subject_image_file).name).split('_')[0] + '_' + \
+                   str(Path(subject_image_file).name).split('_')[1]
+    else:
+        subject_image_file = os.path.join(root, subject, 'anat',
+                                          f"{subject}_{contrast}.nii.gz")
+        subject_label_file = os.path.join(root, 'derivatives', 'labels', subject, 'anat',
+                                          f"{subject}_{contrast}_{label_suffix}.nii.gz")
+        sub_name = str(Path(subject_image_file).name).split('_')[0]
+
+    if os.path.exists(subject_image_file):
+        if label_suffix is not None:
+            if os.path.exists(subject_label_file):
+                subject_label_file_nnunet = os.path.join(path_out_labels,
+                                                         f"{sub_name}_{counter:03d}.nii.gz")
+                list_labels.append(subject_label_file_nnunet)
+                # copy the files to new structure using symbolic links (prevents duplication of data and saves space)
+                os.symlink(os.path.abspath(subject_label_file), subject_label_file_nnunet)
+            else:
+                print(f"Label for image {subject_image_file} does not exist this file is ignored")
+        subject_image_file_nnunet = os.path.join(path_out_images,
+                                                     f"{sub_name}_{counter:03d}_{channel:04d}.nii.gz")
+        list_images.append(subject_image_file_nnunet)
+        # copy the files to new structure using symbolic links (prevents duplication of data and saves space)
+        os.symlink(os.path.abspath(subject_image_file), subject_image_file_nnunet)
+    else:
+        print(f"contrast {contrast} for subject {sub_name} does not exist this contrast is ignored")
+
+    return list_images, list_labels
+
+
 def binarize_label(subject_path, label_path):
+    # For soft seg ground truth nnUNet only accept integer
```

Comment:
This comment is unclear - so what do we do when we have soft seg? I suggest you change this to - "nnUNet only accepts binarized labels, hence the soft labels have to be binarized as done below"

Reply:
nnUNet accept different label but it must be integer value. So I was wondering if by discretizing soft seg value to integer value we can use soft seg?

Reply:
Good point! But the soft segs are usually values between 0-1. So, if you binarize them you would only end up with 0s and 1s (and because of that the softness is lost). 

But, if we can find a way to encode the soft segmentations in a different range (not so 0-1) that can result in integer value after binarizing, yes, we can indeed use soft segs with nnuent. Hope this makes sense!